core/src/test/java/org/infinispan/globalstate/GlobalStateBackwardsCompatibilityTest.java

```diff
@@ -0,0 +1,85 @@
+package org.infinispan.globalstate;
+
+import static org.infinispan.commons.test.CommonsTestingUtil.tmpDirectory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.topology.PersistentUUID;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+/**
+ * ISPN-12667 A test to ensure that global state properties from previous Infinispan majors are compatible.
+ *
+ * @author Ryan Emerson
+ * @since 12.0
+ */
+public class GlobalStateBackwardsCompatibilityTest extends MultipleCacheManagersTest {
+   private static final String CACHE_NAME = "testCache";
+   private static final String MEMBER_0 = PersistentUUID.randomUUID().toString();
+   private static final String MEMBER_1 = PersistentUUID.randomUUID().toString();
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      createCacheManagerWithGlobalState(MEMBER_0, tmpDirectory(this.getClass().getSimpleName(), "0"));
+      createCacheManagerWithGlobalState(MEMBER_1, tmpDirectory(this.getClass().getSimpleName(), "1"));
+      waitForClusterToForm(CACHE_NAME);
+   }
+
+   @AfterClass(alwaysRun = true)
+   protected void clearTempDir() {
+      Util.recursiveFileRemove(tmpDirectory(this.getClass().getSimpleName()));
```

Comment:
-1 to delete the folder in a separate method.
It is being invoked before stopping the cache manager and I'm getting `Caused by: java.io.FileNotFoundException: /home/pedro/Work/infinispan/core/target/infinispanTempFiles/GlobalStateBackwardsCompatibilityTest/1/___global.state (No such file or directory)`

IMO, it is better to overwrite `destroy()`, stopping the cache manager first, and then remove the folder.
```
   @AfterClass(alwaysRun = true)
   @Override
   protected void destroy() {
      super.destroy();
      Util.recursiveFileRemove(tmpDirectory(this.getClass().getSimpleName()));
   }
```